widgetssdk/src/main/java/com/glia/widgets/chat/ChatController.java

```diff
@@ -319,9 +322,9 @@ public void sendMessagePreview(String message) {
         }
     }
 
-    public boolean sendMessage(String message) {
+    public boolean sendMessage(String message, int numOfAttachment) {
```

Comment:
I don't think that adapterItemCount is required here

You could move the validity check to the sendMessageUseCase and introduce separate CallbackListener for that one

In order to check whether it has attachments to send you could ask this info from a AttachmentsProvider

Reply:
this way we will keep the controller clean :)

the sendMessageUseCase has 3 execution methods

basically one should remain execute(String message, CustomCallback callbacks)
with callbacks handling the success case, operator missing error etc. - you don't have to map the error case yet as it will be separate task in the future

others you can make private for now and map to them from this new execute(String message, CustomCallback callbacks)

as this method should be used in case of empty message:
public void execute(FilesAttachment filesAttachment, Listener listener)

Reply:
I think if the message is here the count can be here too, logic is not so different. The controller is a huge mess anyway and this is a very minor thing to improve. If we want to move the count we need to move the logic for the message too which I do not want to atm. Let's leave it as it is since it works otherwise we will be dealing some other issues